testnet/stacks-node/src/event_dispatcher.rs

```diff
@@ -430,11 +432,7 @@ impl EventObserver {
             Self::send_payload_directly(&payload, &url, timeout);
 
             #[cfg(test)]
-            if TEST_EVENT_OBSERVER_SKIP_RETRY
-                .lock()
-                .unwrap()
-                .unwrap_or(false)
-            {
+            if TEST_EVENT_OBSERVER_SKIP_RETRY.with(|v| *v.borrow()) {
```

Comment:
Pervasively, instead of testing a global variable container for a global that should only be compiled for test code, can you create a helper function with an informative name, and a test and non-test body?  Like this:

```rust
use std::sync::LazyLock;
use crate::run_loop::neon::TestFlag;

const TEST_EVENT_OBSERVER_SKIP_RETRY : LazyLock<TestFlag> = LazyLock::new(Testflag::default);

/// Add a helpful docstring here
#[cfg(any(test, feature = "testing"))]
fn fault_injection_skip_observer_entry() -> bool {
   TEST_EVENT_OBSERVER_SKIP_RETRY.get()
}

#[cfg(not(any(test, feature = "testing")))]
fn fault_injection_skip_observer_entry() -> bool {
   false
}
```